units/check_blocked_tasks.go

```diff
@@ -0,0 +1,67 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/task"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+)
+
+const (
+	checkBlockedTasks = "check_blocked_tasks"
+)
+
+func init() {
+	registry.AddJobType(checkBlockedTasks, func() amboy.Job { return makeCheckBlockedTasksJob() })
+}
+
+type checkBlockedTasksJob struct {
+	TaskIDs  []string `bson:"task_ids" json:"task_ids" yaml:"task_ids"`
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+}
+
+func makeCheckBlockedTasksJob() *checkBlockedTasksJob {
+	j := &checkBlockedTasksJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    checkBlockedTasks,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+// NewCheckBlockedTasksJob creates a job to run repotracker against a repository.
+// The code creating this job is responsible for verifying that the project
+// should track push events
+func NewCheckBlockedTasksJob(distroId string, taskIDs []string) amboy.Job {
+	job := makeCheckBlockedTasksJob()
+	job.TaskIDs = taskIDs
+	job.SetID(fmt.Sprintf("%s:%s:%s", checkBlockedTasks, distroId, time.Now().String()))
```

Comment:
Since these jobs aren't added by a cron, I don't think it makes much sense to have a timestamp in them to prevent duplicates. Can we have the ID be some sort of uuid, assuming that all jobs that get added will end up doing something?

Reply:
Job IDs still need to be unique though, right?

Reply:
I think timestamp is as good for uniqueness as anything.

Reply:
it runs the risk of hitting a duplicate key error if 2 tasks with the same distro need to check blocked tasks at a sufficiently close time

Reply:
This saves all the task IDs for the distro on this run of the scheduler so I don't know that that's true; also doesn't time.Now use fractions of seconds?

Reply:
It does, but that doesn't make it random. There's a good reason object IDs append the timestamp with random string. Also the main reason timestamps are in IDs in amboy jobs is to make the cron jobs only have 1 per interval, where the other 19 servers get duplicate key errors